### PR TITLE
[imgui] Split bindings to multiple features

### PR DESCRIPTION
**Pull request by RT2Code**
See #12250.

I tested all the features on x86-windows, x64-windows, x64-windows-static and x64-linux triplets. Everything build fine except:
- dx12-binding on x86-windows. There's an issue opened about it here : https://github.com/ocornut/imgui/pull/2406
- allegro5-binding on x64-linux, because the current allegro5 port fails to build on this platform.

Contrary to the previous port which simply copied the sources to a specific folder, bindings are now compiled with their dependencies. The only exception is Marmelade, because there's no port available for it yet.

Unfortunately, I can't test the OSX and metal bindings. If someone using this platform could test it, it would be very much appreciated. :)

### CODE REVIEW

**Comment by JackBoosY** [comment]
Imgui officials tend to set this port as header only and export it as an interface port.
Therefore:
1. many ports that depend on it use its `.h` and `.cpp` directly to use its functions.
2. imgui did not declare any symbols to be exported in the header file.

So we have to be more careful about imgui changes.

File: ports/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,49 @@
+vcpkg_check_linkage(ONLY_STATIC_LIBRARY)
```
Comment (ras0219-msft) [inline ports/portfile.cmake:1]: I think these are in the wrong directories :)

**Review by ras0219-msft — CHANGES_REQUESTED** [review]

File: ports/portfile.cmake (hunk 1 above)
Comment (RT222) [reply ports/portfile.cmake]: Whoops, I was probably more tired than I thought when I made this PR. Fixed.

File: ports/imgui/CONTROL [hunk 2]
```diff
@@ -1,11 +1,70 @@
 Source: imgui
-Version: 1.77
+Version: 1.77-1
```
Comment (JackBoosY) [inline ports/imgui/CONTROL:2]:
```suggestion
Version: 1.77
Port-Version: 1
```

**Review by JackBoosY — CHANGES_REQUESTED** [review]

File: ports/imgui/CMakeLists.txt [hunk 3]
```diff
@@ -22,20 +22,102 @@ target_sources(
 		${CMAKE_CURRENT_SOURCE_DIR}/misc/cpp/imgui_stdlib.cpp
 )
 
-if(IMGUI_FREETYPE)
-	find_package(Freetype REQUIRED)
+if(IMGUI_BUILD_ALLEGRO5_BINDING)
+	find_path(ALLEGRO5_INCLUDE_DIRS allegro5/allegro.h)
+	target_include_directories(${PROJECT_NAME} PRIVATE ${ALLEGRO5_INCLUDE_DIRS})
+	target_sources(${PROJECT_NAME} PRIVATE ${CMAKE_CURRENT_SOURCE_DIR}/examples/imgui_impl_allegro5.cpp)
```
Comment (JackBoosY) [inline ports/imgui/CMakeLists.txt:28]: I'm not sure if these sample source files should be added to the imgui target.

**Review by JackBoosY — CHANGES_REQUESTED** [review]

**Comment by ras0219-msft** [comment]
>I'm not sure if these sample source files should be added to the imgui target.

Due to the somewhat unique structure of `imgui`, I think this is the best approach we have. The other reasonable alternative would be to turn every backend into a separate port + separate static lib, which would nicely enable adding and removing backends without recompiling everyone using imgui, but that is a lot of boilerplate for something mostly equivalent.

**Comment by ras0219-msft** [comment]
LGTM, thanks for the awesome improvement!

**Pull request merged**